### ISSUE TITLE
Add a button to email PDF of metadata

### ISSUE THREAD

**Issue by cole-fields** [genkimaps/gis-hub#110]
@genkimaps I wrote a little script (https://gitlab.com/dfo-msea/gis-hub/metadata-pdf) to export the GIS Hub metadata to a PDF file. This may be useful to integrate on the GIS Hub as a button (similar to the Export HNAP button, but simpler). We would need to install LaTeX on the GIS Hub server for the script to run. I'm still working on cleaning up the conversion to PDF since there are some markdown characters that mess with the formatting in LaTeX (https://gitlab.com/dfo-msea/gis-hub/metadata-pdf/-/issues/1). 

- [x] Move script to hub-geo-api
- [x] Update the code to make pdf in hub-geo-api to use a default export location, if the output folder is not provided as a parameter to the script. Note: default for export HNAP is `/home/dfo/hub-geo-api/metadata_archive/hnap_metadata`
- [x] need to set output directory in pdflatex call https://tex.stackexchange.com/questions/268997/pdflatex-seems-to-ignore-output-directory/269140 which is better than os.chdir for *.aux and *.log files produced by subprocess
- [ ] Use run_command_as in dfo_plugin_settings to run the script from the command line
- [ ] Add API action in plugins.py to generate the PDF file.
- [ ] Try connecting pdf export function with custom route defined in plugins.py
- [ ] get_pdf function to email document to user

For Cole's reference: https://github.com/genkimaps/gis-hub/issues/91

Examples:

[model_based_density__2021-06-11.pdf](https://github.com/genkimaps/gis-hub/files/6641262/model_based_density__2021-06-11.pdf)
[species_distribution_2021-06-11.pdf](https://github.com/genkimaps/gis-hub/files/6641263/species_distribution_2021-06-11.pdf)
[sponge_reef_areas_2021-06-11.pdf](https://github.com/genkimaps/gis-hub/files/6641264/sponge_reef_areas_2021-06-11.pdf)

**Comment by cole-fields** [comment]
need to set output directory in pdflatex call https://tex.stackexchange.com/questions/268997/pdflatex-seems-to-ignore-output-directory/269140 which is better than os.chdir for *.aux and *.log files produced by subprocess

**Comment by cole-fields** [comment]
@genkimaps I've been chipping away at the code to generate the PDF file and I think it's in a good place. The code can produce 2 versions of the metadata - one without any maps, and one with the spatial resources plotted and included in the PDF. 
[species_distribution_models_northern_and_southern_shelf_bioregion_species_csas_2020_004.pdf](https://github.com/genkimaps/gis-hub/files/6971872/species_distribution_models_northern_and_southern_shelf_bioregion_species_csas_2020_004.pdf)
[species_distribution_models_northern_and_southern_shelf_bioregion_species_csas_2020_004_maps.pdf](https://github.com/genkimaps/gis-hub/files/6971873/species_distribution_models_northern_and_southern_shelf_bioregion_species_csas_2020_004_maps.pdf).

There are occasions where myself or others want to share the metadata with non-DFO staff and just saving the metadata from the GIS Hub as PDF from the browser isn't great. I think there are a few options for including the PDF metadata on the Hub:
1. include in the zip file - both versions can be included. but would have to re-create/upload after every update.
2. host files on something like GitLab - cannot really mix private and public and some of the PDFs with maps should be private
3. a) Create a separate dataset with the zip of the PDF(s) for each existing dataset - easier to update, and can set same permissions as the spatial dataset. 
    b) Create a button as described above that creates the PDF without maps (much faster). And have the PDFs with maps linked as separate resources. For some datasets, creating the PDF with maps takes a long time and so it has to be something pre-made that wouldn't be suitable for a button. 

Is there a better approach?

Code is: https://github.com/genkimaps/hub-geo-api/blob/dataset-pdf-map/dataset-pdf.py and https://github.com/genkimaps/hub-geo-api/blob/dataset-pdf-map/reference/dataset-template.tex

**Comment by cole-fields** [comment]
@genkimaps Just a note that I think the simplest approach is to just create new datasets with the PDFs and link them to their associated datasets. 

This will let us control permissions easily to match the dataset the metadata documents correspond to. I've started the process of uploading these (about 1/3 of the way through). 

Note: I've temporarily disabled emails for the resource permissions as well as the cron job that runs hourly to email users about new datasets. Since for these metadata datasets, i don't really think it's necessary to send emails. I will re-enable this once I'm done (a couple of days i think).

**Comment by genkimaps** [comment]
@cole-fields 
Sorry I'm a bit late to this issue. The code for PDF export looks awesome.  I think your solution is fine, I guess the biggest pitfall with this is that you don't necessarily want to have the PDF stuck inside the zip file, correct?  Why would you not want to have one or more PDFs as additional resources, outside of the resources created from the zip file contents? I think that would work with the hub-geo-api code, as long as you manage those PDFs manually.

**Comment by cole-fields** [comment]
@genkimaps thanks! Over the past few months, we've had several requests for the metadata to share with collaborators/clients outside of DFO and so hopefully this will be a nice, clean solution to share with them. I've finished uploading all the PDFs and linking them/setting permissions except for a few datasets that are in 'Draft' status. 

Shouldn't be much work on my end to maintain these since updates are relatively infrequent. The other approach I initially thought of was to have an export/email button similar to the HNAP functionality. But I think this is fine and won't be much work now that it's been done for most of the datasets.

**Comment by cole-fields** [comment]
Joanne was a bit surprised by the number of datasets doubling but I explained this was a fairly easily way of managing the PDF metadata and having them linked to their respective datasets. I think I'll close this for now and can revisit if we decide to integrate the PDFs in a different way in the future.

**Comment by cole-fields** [comment]
I've updated the datasets page to essentially 'hide' the PDF metadata datasets (see https://github.com/genkimaps/gis-hub/commit/5b09ccf8d91275aa6413fe3fa0cdb9173a138b16). It looked a bit overwhelming with them being displayed in the datasets page. Here, the links from a dataset's resources to the pdf metadata page still work, but the PDF metadata datasets will not show up on /datasets or through the search functionality. one downside is the misleading count of datasets for each organization.

**Comment by cole-fields** [comment]
Reopened because: 
 - dataset count is then misleading/incorrect
 - I did a batch dataset patch for all the pdf-meta records and then they're displayed first in the query results. so there are pages of empty/hidden datasets until user actually gets to populated/visible dataset records

**Comment by cole-fields** [comment]
Moving to separate issue #120

**Comment by genkimaps** [comment]
@cole-fields  Just to clarify: the script you have on Gitlab for creating PDFs is being run manually, correct?  It seems that this is not necessarily something you want to run manually.  I can see that you might not want all datasets to have a PDF version on GIS Hub.  If you do want this to run automatically, I could integrate it with the hub-geo-api backend script.

**Comment by cole-fields** [comment]
@genkimaps - correct, I run the script manually. I actually moved it over to hub-geo-api here: https://github.com/genkimaps/hub-geo-api/blob/dataset-pdf-map/dataset_pdf.py on the dataset-pdf-map branch. I think all of the spatial datasets (except for datasets in Draft status) should have a linked pdf metadata dataset now. I was able to do a batch run of all the datasets, and then manually run a few after I found some characters that LaTeX had problems with. 

I run the script on my desktop, since it requires latex to be installed (https://github.com/genkimaps/hub-geo-api/blob/120240f5d0a093f755403cf4c3077c99df2c4cef/dataset_pdf.py#L881). It also requires geopandas and matplotlib for plotting the vector and raster layers and including them in the PDF. 

One small issue with integrating it on the hub-geo-api script is that the initial creation of a dataset is usually set to draft status and is completed over more than 1 session. Perhaps if it were to be triggered only if the status field is not equal to 'draft'. This would be nice to have, but low priority.
